src/java/grl/src/grl/driver/GRL_Driver.java

```diff
@@ -232,7 +262,9 @@ public void run()
 		while (!stop && !_startStopUI.is_stopped() && _lbr.getSafetyState().getEmergencyStopInt()==EmergencyStop.INACTIVE) {
 			message_counter+=1;
 			_currentKUKAiiwaState = udpMan.waitForNextMessage();
-			_previousKUKAiiwaState = udpMan.getPrevMessage();
+			
+			//Some bug in this, just set _previousKUKAiiwaState = _currentKUKAiiwaState before looping/continue?
+			_previousKUKAiiwaState = udpMan.getPrevMessage();  
```

Comment:
What about teach mode, FRI mode, shutting down, etc? Those are supported in master. By the way do you have a touch flange on your physical kuka?